doc/guides/adding-new-napi-api.md

```diff
@@ -0,0 +1,50 @@
+# Contributing a new API to N-API
+
+N-API is Node.js's next generation ABI-stable API for native modules.
+While improving the API surface is encouraged and welcomed, the following are
+a set of principles and guidelines to keep in mind while adding a new
+N-API API.
+
+* A new API **must** adhere to N-API API shape and spirit.
+    * **Must** be a C API.
+    * **Must** not throw exceptions.
+    * **Must** return `napi_status`.
+    * **Should** consume `napi_env`.
+    * **Must** operate only on primitive data types, pointers to primitive
+      datatypes or opaque handles.
+    * **Must** be a necessary API and not a nice to have. Convenience APIs
+      belong in node-addon-api.
+    * **Must** not change the signature of an existing N-API API  or break

```

Comment:
Extra space before "or"

Reply:
Fixed- thanks! And yes, whoops- the package-lock changes were accidental 😞